spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/properties/KafkaConsumerProperties.java

```diff
@@ -84,47 +84,120 @@ public long getReferencePoint() {
 
 	}
 
+	/**
+	 * When autoCommitOffset is true, this setting dictates whether to commit the offset after each record is processed.
```

Comment:
instead of after the complete batch of records returned by a poll() is processed.